File: app/src/main/java/com/example/techfix/data/DataSeeder.java
package com.example.techfix.data;
import com.example.techfix.data.entity.*;
public final class DataSeeder {
 private DataSeeder(){}
 public static void seedIfEmpty(AppDatabase db){
  if(!db.branchDao().getAllBranches().isEmpty()) return;
  db.branchDao().insertBranch(new Branch("Colombo Branch","123 Galle Road, Colombo",6.9271,79.8612));
  db.branchDao().insertBranch(new Branch("Galle Branch","45 Lighthouse Street, Galle",6.0535,80.2210));
  db.deviceCategoryDao().insertCategory(new DeviceCategory(1,"Smartphone"));
  db.deviceCategoryDao().insertCategory(new DeviceCategory(2,"Laptop"));
  db.deviceCategoryDao().insertCategory(new DeviceCategory(3,"Tablet"));
  db.repairServiceDao().insertService(new RepairService(1,"Screen Replacement","Replace cracked or unresponsive screen",8500.0));
  db.repairServiceDao().insertService(new RepairService(1,"Battery Replacement","Replace degraded battery",4500.0));
  db.repairServiceDao().insertService(new RepairService(2,"Keyboard Repair","Fix unresponsive or damaged keys",6000.0));
  db.repairServiceDao().insertService(new RepairService(2,"Motherboard Diagnostic","Diagnose hardware faults",3000.0));
  db.repairServiceDao().insertService(new RepairService(3,"Screen Replacement","Replace cracked tablet screen",9500.0));
 }
}

File: app/src/main/java/com/example/techfix/data/DataSeeder.kt
package com.example.techfix.data

import com.example.techfix.data.entity.Branch
import com.example.techfix.data.entity.DeviceCategory
import com.example.techfix.data.entity.RepairService

object DataSeeder {

    suspend fun seedIfEmpty(db: AppDatabase) {
        val branchDao = db.branchDao()
        val categoryDao = db.deviceCategoryDao()
        val serviceDao = db.repairServiceDao()

        if (branchDao.getAllBranches().isNotEmpty()) return // already seeded

        branchDao.insertBranch(
            Branch(name = "Colombo Branch", address = "123 Galle Road, Colombo", latitude = 6.9271, longitude = 79.8612)
        )
        branchDao.insertBranch(
            Branch(name = "Galle Branch", address = "45 Lighthouse Street, Galle", latitude = 6.0535, longitude = 80.2210)
        )

        categoryDao.insertCategory(DeviceCategory(categoryId = 1, name = "Smartphone"))
        categoryDao.insertCategory(DeviceCategory(categoryId = 2, name = "Laptop"))
        categoryDao.insertCategory(DeviceCategory(categoryId = 3, name = "Tablet"))

        serviceDao.insertService(RepairService(categoryId = 1, name = "Screen Replacement", description = "Replace cracked or unresponsive screen", price = 8500.0))
        serviceDao.insertService(RepairService(categoryId = 1, name = "Battery Replacement", description = "Replace degraded battery", price = 4500.0))
        serviceDao.insertService(RepairService(categoryId = 2, name = "Keyboard Repair", description = "Fix unresponsive or damaged keys", price = 6000.0))
        serviceDao.insertService(RepairService(categoryId = 2, name = "Motherboard Diagnostic", description = "Diagnose hardware faults", price = 3000.0))
        serviceDao.insertService(RepairService(categoryId = 3, name = "Screen Replacement", description = "Replace cracked tablet screen", price = 9500.0))
    }
}